doc/nrf/ug_thingy91_gsg.rst

```diff
@@ -85,356 +85,385 @@ The data is transmitted to nRF Cloud.
 You must also update the modem firmware.
-You can do this through Universal Serial Bus (USB) (MCUboot) or an external debug probe using the Programmer application.
+You can do this through :term:`Universal Serial Bus (USB)` (MCUboot) or an external debug probe by using the Programmer application.
 
 .. note::
    To update the Thingy:91 through USB, the nRF9160 SiP and nRF52840 SoC bootloaders must be factory-compatible.
    The bootloaders might not be factory-compatible if the nRF9160 SiP or nRF52840 SoC has been updated with an external debug probe.
    To restore the bootloaders, program the nRF9160 SiP or nRF52840 SoC with the Thingy:91 firmware files through an external debug probe.
 
-.. _programming_usb:
 
-Updating firmware through USB
-=============================
+.. note::
+   The external debug probe must support Arm Cortex-M33, such as the nRF9160 DK.
+   You need a 10-pin 2x5 socket-socket 1.27 mm IDC (:term:`Serial Wire Debug (SWD)`) JTAG cable to connect to the external debug probe.
+
+Following are the three recommended steps when you start to program a Thingy:91:
+
+1. :ref:`Updating the Connectivity bridge application firmware in the nRF52840 SoC <updating_the conn_bridge_52840>`.
+#. :ref:`Updating the modem firmware on the nRF9160 SoC (especially if you want to incorporate the latest changes from a newly released modem firmware) <update_modem_fw_nrf9160>`.
+#. :ref:`Programming the application firmware on the nRF9160 SiP <update_nrf9160_application>`.
 
-You can update the Thingy:91 application and modem firmware over :term:`Universal Serial Bus (USB)` by using MCUboot, which is a secure bootloader that can be used to update applications without an external debugger.
 
 Before you start, make sure the Thingy:91 is connected to the computer with a micro-USB cable.
 
 .. note::
 
    Do not unplug the Nordic Thingy:91 during this process.
 
-To update the firmware, complete the following steps:
+.. rst-class:: numbered-step
 
-1. Open `nRF Connect for Desktop`_ and launch the Programmer application.
-#. Scroll down in the menu on the left and make sure :guilabel:`Enable MCUboot` is selected.
+.. _updating_the conn_bridge_52840:
 
-      .. figure:: /images/programmer_enable_mcuboot.png
-         :alt: Programmer - Enable MCUboot
+Updating the firmware in the nRF52840 SoC
+=========================================
 
-         Programmer - Enable MCUboot
+.. tabs::
 
-#. Update the nRF52840 :term:`System on Chip (SoC)` application:
+   .. group-tab:: Through USB
 
-   a. Switch off the Thingy:91.
-   #. Press **SW4** while switching **SW1** to the **ON** position.
+      To update the firmware, complete the following steps:
 
-      .. figure:: /images/thingy91_sw1_sw4.svg
-         :alt: thingy91_sw1_sw4
+      1. Open `nRF Connect for Desktop`_ and launch the Programmer application.
+      #. Scroll down in the menu on the left and make sure :guilabel:`Enable MCUboot` is selected.
```

Comment:
Should these 2 steps be added to the "Update the modem firmware" and "Programming the application firmware" sections also, as there could be a scenario where one could directly navigate to any of these sections without completing the steps in the "Updating the firmware in the nRF52840 SoC" section?